### ISSUE TITLE
Add-AzureRmAccount :: Incorrect credentials aren't throwing an exception

### ISSUE THREAD

**Issue by pcgeek86** [Azure/azure-powershell#1664]
@markcowl 

I just installed AzureRM on a fresh Windows 10 VM, which has never had the module installed before. I am trying to call `Add-AzureRmAccount`, and I'm getting an `AzureProfile` instance back, with no subscriptions tied to it. Thus, I am receiving the error message:

> No subscription found in the context.  Please ensure that the credentials you provided 
are authorized to access an Azure subscription, then run Login-AzureRMAccount to login.

There is also no `Clear-AzureRmProfile` command, so I can start over from scratch, without closing the PowerShell session. What happened to this command?

Cheers,
Trevor Sullivan
Microsoft MVP: PowerShell
http://trevorsullivan.net
http://twitter.com/pcgeek86

<img width="1392" alt="screen shot 2016-01-14 at 8 09 04 pm" src="https://cloud.githubusercontent.com/assets/466713/12344478/b26e0d76-bafa-11e5-8f1d-fa36d2c68288.png">

<img width="381" alt="screen shot 2016-01-14 at 8 09 50 pm" src="https://cloud.githubusercontent.com/assets/466713/12344484/ce0c8d50-bafa-11e5-9039-ec8a5578fb0d.png">

**Comment by pcgeek86** [comment]
If I login with my Microsoft Account (MSA), I get my subscription details back. Is AAD having a temporary problem? Or is this an issue with ARM PowerShell?

**Comment by pcgeek86** [comment]
I just realized that I had the wrong password, because Azure Active Directory (AAD) forced me to change my password a few days ago. Everything is working fine now, except .....

**Shouldn't I be getting a massive, red error message telling me that my credentials were incorrect?**

I just wasted the past half hour trying to figure out what happened. 

Cheers,
Trevor Sullivan

**Comment by markcowl** [comment]
@pcgeek86 Thanks for the report, we'll have to repro this to see what exception is coming back from AAD in this case.  That message normally occurs when you successfully get a token against the common tenant, but you can't authenticate against other tenants, or no subscriptions are found in them.

To confirm, you are using orgid credentials in this case, correct?

**Comment by pcgeek86** [comment]
@markcowl Yes sir, I'm using my Azure Active Directory (AAD) user account, named `trevor@trevorsullivan.net`. It's just weird that it seems to successfully authenticate, using my old password, but doesn't return any subscriptions.

**Comment by hovsepm** [comment]
@pcgeek86 thanks for reporting. I can repro it simply by closing the ADAL dialog when it prompts out. That will cause us returning a Profile without any subscription in it. All the following calls will cause "No subscription found in the context." message. We will fix this in the upcoming release.

**Comment by pcgeek86** [comment]
@hovsepm Thanks for the repro -- yep, I replicated the same "close the ADAL dialog with the X" behavior last night, too. Looking forward to the next release ... do we have tests in place, that would catch this kind of thing before it goes out?

**Comment by hovsepm** [comment]
@pcgeek86 test added with the fix.

**Comment by pcgeek86** [comment]
@hovsepm Excellent, thank you! I was more curious about general standards for testing in the Azure PowerShell project.

Since this is a commercial product / service, that customers are relying on for both production and non-production systems, the introduction of new bugs / breaking changes like this have a serious impact on scripts. For example, what if I wrote a script that used a `try..catch` block to catch invalid authentication? If I upgraded the AzureRm PowerShell module, then it would have broken my script code.

Increasing user confidence is one, of many, things that can increase consumption of the Azure service.
